src/cli/rustup_mode.rs

```diff
@@ -904,6 +906,32 @@ fn update(cfg: &mut Cfg, m: &ArgMatches<'_>) -> Result<utils::ExitCode> {
     if let Some(names) = m.values_of("toolchain") {
         for name in names {
             update_bare_triple_check(cfg, name)?;
+
+            let toolchain_has_triple = match PartialToolchainDesc::from_str(name) {
+                Ok(x) => x.has_triple(),
+                _ => false,
+            };
+
+            if toolchain_has_triple {
+                let host_arch = TargetTriple::from_host_or_build();
+                match ToolchainDesc::from_str(name) {
+                    Ok(toolchain_desc) => {
+                        let target_triple = toolchain_desc.target.clone();
+                        if host_arch.ne(&target_triple) {
```

Comment:
Why did you feel the need to clone that here?

Reply:
fixed